lib/matplotlib/ticker.py

```diff
@@ -2496,9 +2496,9 @@ def tick_values(self, vmin, vmax):
         # We could also add ticks at t, but that seems to usually be
         # uninteresting.
         #
-        # "simple" mode is when the range falls entirely within (-t,
-        # t) -- it should just display (vmin, 0, vmax)
-        if -linthresh < vmin < vmax < linthresh:
+        # "simple" mode is when the range falls entirely within [-t, t]
+        #  -- it should just display (vmin, 0, vmax)
```

Comment:
I feel like we're still not honouring this comment completely: shouldn't we include 0 too if `vmin<0 and vmax>0` ?

Reply:
instead of nesting conditionals, we could simply return `sorted(set([vmin, 0, vmax]))`. WDYT @rcomer ?

Reply:
Good point!  I had been assuming the zero gets added in somewhere else, but if I subtract 0.5 from the data in your example I get this (with `main` branch).
![temp](https://github.com/matplotlib/matplotlib/assets/10599679/5839218e-853b-49c4-8c9d-f2f210288237)

If I then reduce linthresh to 0.1 (so we are no longer in "simple" mode) we do see the zero:
![temp](https://github.com/matplotlib/matplotlib/assets/10599679/76317482-3ba2-4c87-8cb5-33db1aac656c)

So for consistency, it seems to me that the comment is right and a zero should be included.  I think it's up to you whether you want to change it though.  You could decide it's beyond the scope of this PR.

Reply:
> instead of nesting conditionals, we could simply return `sorted(set([vmin, 0, vmax]))`. WDYT @rcomer ?

I did not see the before my last comment but, yes, that makes sense to me.

Reply:
great, thank you for testing this case, I was also assuming the 0 might be added somewhere else.
I fixed this too and added a test case !